Issue #32981 || Corrected Spinbutton and Slider role "only read-write" content and did some changes on both so that both roles can be more consistent and similar. (#36488)
mdn/content: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md

### ARIA range widget options

> ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, scrollbars, range input, sliders and spinbuttons.

Reviewer: again a range input is not an aria role.  a range input (html) is the same as a role=slider.

```suggestion
ARIA provides developers with five different [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles) that inherit from the range abstract role, including progressbars, meters, scrollbars, spinbuttons and sliders.
```

Author: Same changes are implemented

> The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction, such as a file upload's loading progress bar that eventually reaches 100% when fully loaded.

Reviewer: ```suggestion
The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction. For instance, a file upload's progress bar that conveys the data transfer up to 100%, when the file has fully uploaded.
```

Author: Requested changes are implemented

> The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only gauge. It indicates the amount of something within a known range, such as a computer's battery indicator or a car's gas gauge.

Reviewer: ```suggestion
The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only range. It represents a scalar measurement within a known range, or a fractional value. For instance, a computer's battery indicator, the disk usage of a hard drive or a car's gas gauge.
```

Author: Suggested changes are implemented.

> The `slider` role, similar to HTML's `input` of type `range`, [`<input type="range">`](/en-US/docs/Web/HTML/Element/input/range), is a read-write input range. Sliders allow users to select a value between the predefined minimum and maximum values. The user selects a value by moving a slider thumb along a horizontal or vertical slider to select a value.

Reviewer: ```suggestion
The `slider` role, similar to HTML's `input` of type `range`, [`<input type="range">`](/en-US/docs/Web/HTML/Element/input/range), is a read-write input range. Sliders allow users to select a value between predefined minimum and maximum values. The user selects a value by moving a slider thumb along a horizontal or vertical slider to select a value.
```

Author: Changes are made

> The `spinbutton` role is a read-write range: it's value changes via user interaction. As such, it must be able to receive focus. In addition, keyboard interaction, mouse clicks, and touch interaction must be supported.

Reviewer: this seems like it was pulled from the slider doc and just replaced slider with spinbutton.  The deletion of saying it was the only one whose value changes via user interaction was a good fix, but might as well call out the similarities between slider and spinbutton here.

```suggestion
The `spinbutton` and `slider` roles' values change via user interaction. As such, these roles must be able to receive focus. In addition, to adjust the value keyboard interaction, mouse clicks, and touch interaction must be supported.
```

Author: I'm glad you found the fix helpful. I understood your point about highlighting the similarities between the slider and spinbutton and the changes has been incorporated.

> > [!WARNING]
> > To change the spinbutton value, touch-based assistive technologies need to respond to user gestures for increasing and decreasing the value by synthesizing key events.

Reviewer: This warning seems like it was copied over from the slider doc. It's not really accurate/applicable to most spinbutton implementations, so i suggest removing it.

```suggestion
```

Author: As suggested the less applicable warning is removed. Please let me know if any further improvements can be done.
